### DISCUSSION TITLE
Laravel migration with more than one timestamp column

### DISCUSSION THREAD

**Discussion by amir-khoshbakht** [laravel/framework#39528] [Q&A] [answered]
"laravel/framework": '8.69.0'
PHP 8.0.11 (cli) (built: Sep 23 2021 21:26:24) ( NTS )
mysql  Ver 15.1 Distrib 10.6.4-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

### Description:

im trying to run this code for migration :
```
            Schema::create('something', function (Blueprint $table) {
                $table->timestamp('a');
                $table->timestamp('b');
                $table->timestamp('c');
            });
```
And I get this error message :

>  SQLSTATE[42000]: Syntax error or access violation: 1067 Invalid default value for 'b' (SQL: create table `something` (`a` timestamp not null, `b` timestamp not null, `c` timestamp not null) default character set utf8mb4 collate 'utf8mb4_unicode_ci')

but when I copy the generated SQL code and run it **mysql** app there is no problem and it works
Exactly the same thing is explained in this link : [https://www.buzzphp.com/posts/laravel-error-in-creating-a-table-with-two-timestamp-columns](url)

### Steps To Reproduce:

1- make a migration with **_more than one timestamp_** column ( $table->timestamp('a'); )
2-migrate

**Comment by lk77** [comment, marked as answer]
you need to set a default value or make the columns nullable

From mysql docs : 
```
the first TIMESTAMP column has both DEFAULT
CURRENT_TIMESTAMP and ON UPDATE
CURRENT_TIMESTAMP
if neither is specified explicitly. 

```

column b and c need  to have a default value or be nullable

**Reply by amir-khoshbakht** [reply]
Why does the same query work in mysql terminal app ?

> you need to set a default value or make the columns nullable

```
//default value
$table->timestamp('a')->useCurrent();
$table->timestamp('a')->default('something...');

//nullable
$table->timestamp('a')->nullable();
```

**Reply by lk77** [reply]
it does not work in mysql terminal

```
create table something (`a` timestamp not null, `b` timestamp not null, `c` timestamp not null)
MySQL responded : Documentation

#1067 - Invalid default value for 'b' 
```

or you have something strange in your setup, perhaps laravel is using a different database than you are in console, with different configurations
